crates/tiny/src/main.rs

```diff
@@ -178,3 +193,41 @@ fn run(
 
     runtime.block_on(local);
 }
+
+// Helper to parse SASL config into a libtiny_client SASL struct
+fn sasl_from_config(tls: bool, sasl_config: &config::SASLAuth) -> Result<SASLAuth, String> {
+    match sasl_config {
+        config::SASLAuth::Plain { username, password } => Ok(SASLAuth::Plain {
+            username: username.clone(),
+            password: password.clone(),
+        }),
+        config::SASLAuth::External { pem } => {
+            // TLS must be on for EXTERNAL
+            if !tls {
+                Err("TLS is not enabled for this server, but SASL EXTERNAL authentication requires SASL. Please enable SASL for this server in the config file.".to_string())
```

Comment:
Could you split this into multiple lines with `\` please? Example: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=93a542110e8ce487180bb770e4af1b84

Reply:
Odd that rustfmt doesn't take care of this. Do we need to use a custom `rustfmt.toml`?